README.md

```diff
@@ -87,6 +87,109 @@ with Maven.
+        <outputDirectory>src/main/avro</outputDirectory>
+        <subjectPatterns>
+            <param>^TestSubject000-(Key|Value)$</param>
+        </subjectPatterns>
+    </configuration>
+</plugin>
+```
+
+## schema-registry:register
+
+This plugin is used to register AVRO schemas for subjects.
+
+|        Name        |                                                      Description                                                      |  Type  | Required | Default |
+|:------------------:|:---------------------------------------------------------------------------------------------------------------------:|:------:|:--------:|:-------:|
+| schemaRegistryUrls | Urls for the schema registry instance to connect to                                                                   |  List  |    yes   |         |
+| subjects           | Map containing subject to schema path of the subjects to be registered.                                               |   Map  |    yes   |         |
+
+```
+<plugin>
+    <groupId>io.confluent</groupId>
+    <artifactId>kafka-schema-registry-maven-plugin</artifactId>
+    <version>3.1.0-SNAPSHOT</version>
+    <configuration>
+        <schemaRegistryUrls>
+            <param>http://192.168.99.100:8081</param>
+        </schemaRegistryUrls>
+        <subjects>
+            <TestSubject000-Key>src/main/avro/TestSubject000-Key.avsc</TestSubject000-Key>
+            <TestSubject000-Value>src/main/avro/TestSubject000-Value.avsc</TestSubject000-Value>
+        <subjects>
+    </configuration>
+    <goals>
+        <goal>register</goal>
+    </goals>
+</plugin>
+```
+
+## schema-registry:test-compatibility
+
+|        Name        |                                                      Description                                                      |  Type  | Required | Default |
+|:------------------:|:---------------------------------------------------------------------------------------------------------------------:|:------:|:--------:|:-------:|
+| schemaRegistryUrls | Urls for the schema registry instance to connect to                                                                   |  List  |    yes   |         |
+| subjects           | Map containing subject to schema path of the subjects to be registered.                                               |   Map  |    yes   |         |
+
+```
+<plugin>
+    <groupId>io.confluent</groupId>
+    <artifactId>kafka-schema-registry-maven-plugin</artifactId>
+    <version>3.1.0-SNAPSHOT</version>
+    <configuration>
+        <schemaRegistryUrls>
+            <param>http://192.168.99.100:8081</param>
+        </schemaRegistryUrls>
+        <subjects>
+            <TestSubject000-Key>src/main/avro/TestSubject000-Key.avsc</TestSubject000-Key>
+            <TestSubject000-Value>src/main/avro/TestSubject000-Value.avsc</TestSubject000-Value>
+        <subjects>
+    </configuration>
+    <goals>
+        <goal>test-compatibility</goal>
+    </goals>
+</plugin>
+```
+
+
```

Comment:
Overall, I think what this might need is a guide for how you might use this in practice, e.g. what stages do you attach each of these to? `test-compatibility` makes sense before a normal compile, I'm not sure when downloading makes sense, and register probably makes sense upon some deployment step (or a custom on you might use via a separate CI step?).